### ISSUE TITLE
Finalizer task runs when finalized task is skipped

### ISSUE THREAD

**Issue by scott-pollom** [gradle/gradle#17347]
### Steps to Reproduce 

1. Add the following to the `build.gradle`:

```
tasks.register("taskX") {
    finalizedBy 'taskY'
    it.enabled(false)
    doLast {
        println 'taskX'
    }
}

tasks.register("taskY") {
    doLast { println 'taskY' }
}
```

2. Run `./gradlew taskX`

### Expected Behavior

I expect neither task to run based on the [user guide](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:finalizer_tasks): "finalizer tasks are not executed if the finalized task didn’t do any work" 

### Current Behavior

Only taskY runs

### Context

https://issuetracker.google.com/issues/189877657

### Your Environment

Gradle 7.0

**Comment by scott-pollom** [comment]
Hi, any update on this bug?

**Comment by mlopatkin** [comment]
This seems to be fixed in 8.8:
https://docs.gradle.org/current/userguide/controlling_task_execution.html#sec:finalizer_tasks
